tests/test_table.py

```diff
@@ -1143,6 +1143,58 @@ def test_load(tmpdir):
         os.remove(f"{path_no_ext}.{ext}")
 
 
+def test_load_broken_csv(tmpdir):
+    r"""Test loading of malformed csv files.
+
+    If csv files contain a lot of special characters,
+    or a different number of columns,
+    than specified in the database header,
+    loading of them should not fail.
+
+    See https://github.com/audeering/audformat/issues/449
+
+    Args:
+        tmpdir: tmpdir ficture
+
+    """
+    build_dir = audeer.mkdir(tmpdir, "build")
+
+    # Create database with single table and column
+    #
+    # Ensure:
+    #
+    # * it contains an empty table
+    # * the columns use schemes with time and date data types
+    # * at least one column has no scheme
+    #
+    # as those cases needed special care with csv files
+    #
+    db = audformat.Database("mydb")
+    db.schemes["date"] = audformat.Scheme("date")
+    db.schemes["time"] = audformat.Scheme("time")
+    db["table"] = audformat.Table(audformat.filewise_index("file.wav"))
+    db["table"]["date"] = audformat.Column(scheme_id="date")
+    db["table"]["date"].set([pd.to_datetime("2018-10-26")])
+    db["table"]["time"] = audformat.Column(scheme_id="time")
+    db["table"]["time"].set([pd.Timedelta(1)])
+    db["table"]["no-scheme"] = audformat.Column()
+    db["table"]["no-scheme"].set(["label"])
+    db["empty-table"] = audformat.Table(audformat.filewise_index())
+    db["empty-table"]["column"] = audformat.Column()
+
+    # Add a hidden column to the table dataframes,
+    # without adding it to the table header
+    db["table"].df["hidden"] = ["hidden"]
+    db["empty-table"].df["hidden"] = []
+
+    db.save(build_dir, storage_format="csv")
```

Comment:
The setup of the test data and putting it into testable state consumes most of the lines implemented and makes it a little harder to read - I mean that the purpose of the test is pretty obvious so maybe this is not crucial at all. 
 
Still I find it stilistically nicer to separate test data creation from asserting .

The pytest way of doing so would very likely to implement (yet another) fixture.
One could though use a different, less pytest-like idiom, test classes containing setUp methods from the built-iin `unittest` module. Would something like [this](https://hands-on.cloud/python-unit-tests/#mocking-classes-in-python-unit-tests:~:text=for%20each%20test.-,Here%20is%20an%20example%3A,-import%20unittest%0Aclass)  not read nicely in the current situation? I mean it looks quite imperative but still separates setup and destruction of test data - like all xunit frameworks do.

Reply:
What I can try to do, is to create a test class and a method to set up the data, and one that does the test.

I will try to do that.

Reply:
OK, sorry, I was too fast with my comment in https://github.com/audeering/audformat/pull/450#discussion_r1675799599 as you can have the data definition and the actual test as two methods of the same class, and hence they are defined together and are easy to read.

I adapted the test to now have the following structure:

```python
class TestLoadBrokenCsv:

    def database_with_hidden_columns(self) -> audformat.Database:
        # create database

    def test_load_broken_csv(self, tmpdir):
        db = self.database_with_hidden_columns()
        # run the test
```

Reply:
>> The creation of the data is the most important part of the test, and the data is meaningless in another test.

>> When creating a fixture, the code for the table is positioned at another place, e.g. at the top of the file, and a user needs to switch between the two positions in order to read it.

I think you could just put it right before the test itself. 

>> But I have to admit, I was never a big fan of object based programming and unittest, so maybe I just don't understand how you would like to have it ;)

In principle what you do is very similar. You could rename the method that creates the data as a @classmethod decorated `setup`  method and you would be almost done (you could use a `teardown` method to delete `tmpdir`
 ). 

But in this particular case this really does not matter, the way you implement it is good for the readability due to data creation and asserting being separated into two methods.   Using a class rather than a function imo also improves readability - you know that the data setup is used in this particular class.  xunit users often would setup and teardown methods, and then go one step further and have each assertion in a single method `def`.  

But for me this is definitively good enough. Not though that due to temporary failures, I was able to run the tests, but I could not recap the data examples from the issue description 

Still I think this is ready to be merged and I will approve.

Reply:
Resolving this thread also.

Reply:
I again cannot find the location where approval is given. But you can go ahead.